js/web/lib/wasm/wasm-training-core-impl.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {InferenceSession} from 'onnxruntime-common';
+
+import {SerializableModeldata, SerializableSessionMetadata} from './proxy-messages';
+import {setSessionOptions} from './session-options';
+import {getInstance} from './wasm-factory';
+import {checkLastError} from './wasm-utils';
+
+const NO_TRAIN_FUNCS_MSG =
+    'Built without training API\'s enabled. Use the onnxruntime-web/training import for training ' +
+    'functionality, and make sure that all the correct artifacts are built & moved to the correct folder if ' +
+    'using a custom build. Check https://onnxruntime.ai/docs/build/web.html for more information.';
+
+export const createCheckpointHandle = (checkpointData: SerializableModeldata): number => {
+  const wasm = getInstance();
+
+  const [checkpointDataOffset, checkpointDataLength] = checkpointData;
+  let checkpointHandle = 0;
+
+  try {
+    if (wasm._OrtTrainingLoadCheckpoint) {
+      checkpointHandle = wasm._OrtTrainingLoadCheckpoint(checkpointDataOffset, checkpointDataLength);
+    } else {
+      throw new Error(NO_TRAIN_FUNCS_MSG);
+    }
+
+    if (checkpointHandle === 0) {
+      checkLastError('Error occurred when trying to create a CheckpointState.');
+    }
+    return checkpointHandle;
+  } catch (e) {
+    if (wasm._OrtTrainingReleaseCheckpoint && checkpointHandle !== 0) {
+      wasm._OrtTrainingReleaseCheckpoint(checkpointHandle);
+    }
+    throw e;
+  } finally {
+    // free buffer from wasm heap
+    wasm._OrtFree(checkpointData[0]);
+  }
+};
+
+const getTrainingModelInputOutputCount = (trainingSessionId: number): [number, number] => {
```

Comment:
these are all internal methods right? You can make this generic
```suggestion
const getModelInputOutputCount = (trainingSessionId: number, IsEvalModel: boolean): [number, number] => {
```